R/ard_stack.R

```diff
@@ -0,0 +1,130 @@
+#' big N's, missingness, attributes) and/or tidy for use in displays with
+#' `shuffle_ard()`.
+#'
+#' @param data (`data.frame`)\cr
+#'   a data frame
+#' @param by ([`tidy-select`][dplyr::dplyr_tidy_select])\cr
+#'   columns to tabulate by in the series of ARD function calls
+#' @param ... ([`dynamic-dots`][dyn-dots])\cr
+#'   Series of ARD function calls to be run and stacked
+#' @param .compute_by_ns (`logical`)\cr
+#'   logical indicating whether to compute big N's for the `by` variable
+#'   Default is `TRUE`.
+#' @param .compute_missing (`logical`)\cr
+#'   logical indicating whether to include the results of `ard_missing()` for all
+#'   variables represented in the ARD. Default is `FALSE`.
+#' @param .add_attributes (`logical`)\cr
+#'   logical indicating whether to include the results of `ard_attributes()` for all
+#'   variables represented in the ARD. Default is `FALSE`.
+#' @param .shuffle (`logical`)\cr
+#'   logical indicating whether to perform `shuffle_ard()` on the final result.
+#'   Default is `FALSE`.
+#' @param .trim (`logical`)\cr
+#'   logical indicating whether to set `trim = TRUE` in the call to `shuffle_ard()`.
+#'   Default is `TRUE`. Will be ignored if `.shuffle = FALSE`.
+#'
+#' @return  a transformed ARD data frame (of class 'card' if `.shuffle = FALSE`)
+#'
+#' @export
+#' @examples
+#'
+#' ard_stack(
+#'   data = ADSL,
+#'   by = "ARM",
+#'   ard_categorical(variables = "AGEGR1"),
+#'   ard_continuous(variables = "AGE")
+#' )
+#'
+#' ard_stack(
+#'   data = ADSL,
+#'   by = "ARM",
+#'   ard_categorical(variables = "AGEGR1"),
+#'   ard_continuous(variables = "AGE"),
+#'   .shuffle = TRUE
+#' )
+#'
+ard_stack <- function(data,
+                      by = NULL,
+                      ...,
+                      .compute_by_ns = TRUE,
+                      .compute_missing = FALSE,
+                      .add_attributes = FALSE,
+                      .shuffle = FALSE,
+                      .trim = TRUE) {
+  # capture quosures -----------------------------------------------------------
+  dots <- enquos(...)
+  by <- enquo(by)
+
+  # check inputs ---------------------------------------------------------------
+  check_not_missing(data)
+  check_class_data_frame(x = data)
+
+  check_class(.compute_by_ns, "logical")
+  check_scalar(.compute_by_ns)
+  check_class(.compute_missing, "logical")
+  check_scalar(.compute_missing)
+  check_class(.add_attributes, "logical")
+  check_scalar(.add_attributes)
+  check_class(.shuffle, "logical")
+  check_scalar(.shuffle)
+  check_class(.trim, "logical")
+  check_scalar(.trim)
+
+  # evaluate the dots using common `data` and `by`
+  ard_list <- lapply(
+    dots,
+    function(x) {
+      x_rhs <- f_rhs(x)
+      x_fn <- call_name(x_rhs)
+      x_args <- call_args(x_rhs)
+      do.call(x_fn, c(list(data = data, by = by), x_args), envir = attr(x, ".Environment"))
+    }
+  )
+
+  # compute Ns by group / combine main calls
+  if (isTRUE(.compute_by_ns) && !quo_is_null(by)) {
```

Comment:
Perhaps simpler to just evaluate the by argument first with the process function, then we're just dealing with a character vector instead of a tidyselect quo